lib/rspec/matchers/built_in/base_matcher.rb

```diff
@@ -53,11 +53,12 @@ def match_unless_raises(*exceptions)
         end
 
         # @api private
-        # Generates a "pretty" description using the logic in {Pretty}.
+        # Generates a description using {EnglishPhrasing}.
         # @return [String]
         def description
-          return name_to_sentence unless defined?(@expected)
-          "#{name_to_sentence}#{to_sentence @expected}"
+          desc = EnglishPhrasing.split_words(name)
+          desc += EnglishPhrasing.list(@expected) if defined?(@expected)
```

Comment:
For perf reasons we can use `<<` here rather than `+=` and allocate one less string.